src/vm/perfinfo.cpp

```diff
@@ -0,0 +1,117 @@
+#include "common.h"
+
+#if defined(FEATURE_PERFMAP) && !defined(DACCESS_COMPILE)
+#include "perfinfo.h"
+#include "pal.h"
+
+PerfInfo::PerfInfo(int pid)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    WCHAR tempPath[MAX_LONGPATH+1];
+    if (!GetTempPathW(MAX_LONGPATH, tempPath))
+    {
+        return;
+    }
+
+    SString path;
+    path.Printf("%Sperfinfo-%d.map", tempPath, pid);
+    OpenFile(path);
+}
+
+void PerfInfo::LogImage(PEFile* pFile, WCHAR* guid)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(pFile != NULL);
+        PRECONDITION(guid != NULL);
+    } CONTRACTL_END;
+
+    SString value;
+    const SString& path = pFile->GetPath();
+    value.Printf("%S%c%S", path.GetUnicode(), sDelimiter, guid);
+
+    SString command;
+    command.Printf("%s", "ImageLoad");
+    WriteLine(command, value);
+
+}
+
+void PerfInfo::WriteLine(SString& type, SString& value)
+{
+
+    CONTRACTL    
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(type != NULL);
+        PRECONDITION(value != NULL);
+    } CONTRACTL_END;
+
+    if (m_Stream == NULL)
+    {
+        return;
+    }
+
+    SString line;
+    line.Printf("%S%c%S%c\n",
```

Comment:
My thought was that the semi-colon is just another thing for someone else to parse out.  It just affects other users of the file.  For example, if someone adds another type of entry, then they need to parse the semi-colon out on the other end just like you do for this scenario.

Not a huge deal either way, so if it makes the parsing simpler, I think it's fine to leave it.